src/routes/data/types.rs

```diff
@@ -11,7 +11,7 @@ use masking::PeekInterface;
 
 use crate::error;
 
-#[derive(serde::Serialize, serde::Deserialize, Debug)]
+#[derive(serde::Serialize, serde::Deserialize, Debug, PartialEq, Eq, Clone)]
```

Comment:
if possible, i don't want big data types to implement clone if it can be avoided